modules/core/src/smithy4s/http/internals/MetaDecode.scala

```diff
@@ -92,14 +91,13 @@ private[http] sealed abstract class MetaDecode[+A] {
         lookupAndProcess(_.query, q) { (values, fieldName, putField) =>
           putField(fieldName, f(values.iterator))
         }
+      // see https://smithy.io/2.0/spec/http-bindings.html#httpqueryparams-trait
+      // when targeting Map[String,String] we take the first value encountered
       case (QueryParamsBinding, StringMapMetaDecode(f)) => {
         (metadata, putField) =>
-          val iter = metadata.query.iterator
-            .filterNot { case (k, _) => reservedQueries(k) }
+          val iter: Iterator[(FieldName, FieldName)] = metadata.query.iterator
             .map { case (k, values) =>
-              if (values.size == 1) {
-                k -> values.head
-              } else throw MetadataError.ArityError(fieldName, QueryBinding(k))
+              k -> values.head
```

Comment:
I wouldn't remove the condition, but rather change it to `if (values.size >= 1) ... else throw new MetadataError.NotFound`